core/src/test/scala/unit/kafka/log/LogSegmentTest.scala

```diff
@@ -65,6 +66,21 @@ class LogSegmentTest {
     Utils.delete(logDir)
   }
 
+  /**
+   * If the maximum offset beyond index, appended to the log section, it throws LogSegmentOffsetOverflowException
+   */
+  @Test
+  def testAppendForLogSegmentOffsetOverflowException(): Unit = {
+    val seg = createSegment(0)
+    val largestOffset: Long = Integer.MAX_VALUE + 1
+    val largestTimestamp: Long = Time.SYSTEM.milliseconds()
+    val shallowOffsetOfMaxTimestamp: Long = 0
+    val memoryRecords: MemoryRecords = records(0, "hello")
+    assertThrows(classOf[LogSegmentOffsetOverflowException], () => {
+      seg.append(largestOffset, largestTimestamp, shallowOffsetOfMaxTimestamp, memoryRecords)
```

Comment:
nit: `largestTimestamp` could be misleading as it could make believe the timestamp itself is set to the maximum possible value.

Reply:
Thanks, I should have changed the variable name to the 'currentTime'